### PR TITLE
RI-327 Configure Octavia for production use

### PR DESCRIPTION
**Pull request by johnsom**
This patch sets the recommended settings production settings for Octavia.

(cherry picked from commit 01fbff1e01248ea37ec98f09e7c557462601ec00)

Issue: [RI-327](https://rpc-openstack.atlassian.net/browse/RI-327)

### CODE REVIEW

File: etc/openstack_deploy/group_vars/all/osa.yml [hunk 1]
```diff
@@ -71,6 +71,50 @@ glance_glance_api_conf_overrides:
...
+# Octavia HA settings
+octavia_loadbalancer_topology: ACTIVE_STANDBY
+octavia_spare_amphora_pool_size: 0
+octavia_enable_anti_affinity: "{{ lookup('env', 'DEPLOY_AIO') != 'yes' }}"
+
+# no event streamer
+octavia_event_streamer: False
+octavia_sync_provisioning_status: False
+
+rackspace_octavia_files_folder: "/opt/rpc-openstack/playbooks/templates/octavia"
```
Comment (prometheanfire) [inline etc/openstack_deploy/group_vars/all/osa.yml]: why is this needed? (and where is it used)?

File: etc/openstack_deploy/group_vars/all/osa.yml [hunk 2]
```diff
@@ -71,6 +71,50 @@ glance_glance_api_conf_overrides:
   DEFAULT:
     image_cache_dir: None
 
+# Octavia tuning
+# activate octavia - Queens only
```
[the remark this answers is not in the document]
Comment (prometheanfire) [reply etc/openstack_deploy/group_vars/all/osa.yml]: rocky/stein

File: etc/openstack_deploy/group_vars/all/osa.yml [hunk 3]
```diff
@@ -71,6 +71,50 @@ glance_glance_api_conf_overrides:
...
+octavia_v2: True
+octavia_v1: False
+# Keep this while there is no Barbican or Vault
+octavia_tls_listener_enabled: False
```
[the remark this answers is not in the document]
Comment (johnsom) [reply etc/openstack_deploy/group_vars/all/osa.yml]: No, barbican has bugs that make it not work with RPC-O, so it's delayed.

File: etc/openstack_deploy/group_vars/all/osa.yml (hunk 1 above)
Comment (johnsom) [reply etc/openstack_deploy/group_vars/all/osa.yml]: Two lines below this one.

File: etc/openstack_deploy/group_vars/all/osa.yml [hunk 4]
```diff
@@ -71,6 +71,50 @@ glance_glance_api_conf_overrides:
...
+
+# Octavia HA settings
+octavia_loadbalancer_topology: ACTIVE_STANDBY
+octavia_spare_amphora_pool_size: 0
+octavia_enable_anti_affinity: "{{ lookup('env', 'DEPLOY_AIO') != 'yes' }}"
```
[the remark this answers is not in the document]
Comment (johnsom) [reply etc/openstack_deploy/group_vars/all/osa.yml]: AIO does not have multiple hosts, so can't use nova anti-affinity to spread the nova instances across hosts.

File: etc/openstack_deploy/group_vars/all/osa.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnsom) [reply etc/openstack_deploy/group_vars/all/osa.yml]: Just removing that legacy comment about the release.

**Comment by johnsom** [comment]
Addressed the comments.

File: etc/openstack_deploy/group_vars/all/osa.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (prometheanfire) [reply etc/openstack_deploy/group_vars/all/osa.yml]: My question was more about what if that env var is not defined. seems like it'll either return yes or null.  pipe into a bool?

[the remark this answers is not in the document]
Comment (johnsom) [reply etc/openstack_deploy/group_vars/all/osa.yml]: This is not my code, so would have to go look this up.  It's what German shipped for K8s and setup for queens....  Even so, null should != 'yes' right?

**Comment by johnsom** [comment]
Same, answering a question...

File: etc/openstack_deploy/group_vars/all/osa.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (johnsom) [reply etc/openstack_deploy/group_vars/all/osa.yml]: Appears to be undocumented in the Ansible docs (of course), but testing shows it's an empty string.

[the remark this answers is not in the document]
Comment (prometheanfire) [reply etc/openstack_deploy/group_vars/all/osa.yml]: ok, since you are comparing it to a quoted string it should be fine then `'yes'`, not `yes`.
